### PR TITLE
Azure: Make the rootdisksize on nodepool configurable

### PR DESCRIPTION
**Pull request by alvaroaleman**
Ref https://issues.redhat.com/browse/HOSTEDCP-323

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: api/v1alpha1/nodepool_types.go [hunk 1]
```diff
@@ -422,4 +422,8 @@ type AgentNodePoolPlatform struct {
 type AzureNodePoolPlatform struct {
 	VMSize  string `json:"vmsize"`
 	ImageID string `json:"imageID"`
+	// DiskSizeGB is the size in GB.
```
Comment (enxebre) [inline api/v1alpha1/nodepool_types.go:425]: thoughts about defaulting to 30GB here and make it explicit and predictable?

Comment (alvaroaleman) [reply api/v1alpha1/nodepool_types.go]: Good idea, updated

**Comment by sjenning** [comment]
Is the minimum (16) or default (30) documented somewhere?  Is this in alignment with standalone OCP minimum/defaults?

**Comment by alvaroaleman** [comment]
> Is the minimum (16) or default (30) documented somewhere? Is this in alignment with standalone OCP minimum/defaults?

Minimum on the api, the default on the CLI (it only is a default if you use the CLI). OCP uses a default of 120G on AWS and 128 on Azure, I can update this to use 120 as well. I am not sure if OCP has a documented minimum, I couldn't find something, are you aware of anything there?

**Comment by alvaroaleman** [comment]
Updated to use 120 as default

**Comment by sjenning** [comment]
I also could not find a minimum.  Agree with default at 120 :+1:

/lgtm

**Comment by enxebre** [comment]
If the agreement is that 120 is a recommended sane default, can we default that in the API as well? The API defines HyperShift, the CLI does not.

**Comment by alvaroaleman** [comment]
> If the agreement is that 120 is a recommended sane default, can we default that in the API as well? The API defines HyperShift, the CLI does not.

Updated, ptal.

**Pull request merged**